lib/opensusebasetest.pm

```diff
@@ -25,21 +25,98 @@ sub post_run_hook {
 
 sub save_and_upload_log {
     my ($cmd, $file, $args) = @_;
-    script_run "$cmd > $file";
-    upload_logs $file;
+    if ($args->{timeout}) {
+        script_run("$cmd | tee $file", $args->{timeout});
+    }
+    else {
+        script_run("$cmd | tee $file");
+    }
```

Comment:
perl beginner question: As `$args` should always exist as a variable, does it work to just write

```perl
script_run("$cmd | tee $file", $args->{timeout});
```
in any case without getting a warning?